src/arithmetic.js

```diff
@@ -64,18 +64,29 @@ function shuntingYard(text) {
         queue.push(stack.shift());
       }
       stack.unshift(token);
+      if (FUNCTION_NAMES.includes(token)) queue.push('END_ARGS');
       continue;
     }
     if (token === '(') {
       stack.unshift(token);
       continue;
     }
+    if (token === ',') {
+      while (shouldPopOperator(token, stack[0])) {
+        queue.push(stack.shift());
+      }
+      if (stack[0] !== '(' || !FUNCTION_NAMES.includes(stack[1])) throw new Error('Incorrect function call');
```

Comment:
I would be slightly better to throw a more specific error if the function has correct syntax, but is not supported: `'${stack[1]} is not a supported function'`

You could update your tests to assert that the errors contain specific text

Reply:
Actually, it is super unclear from the code (Perhaps it should be searched for with a regex instead), but this is to fix this test:
`expect(() => runArithmetic('max((1, 2), 3)')).toThrow();` A non-supported function will not be tokenized correctly and will therefore throw on line 102: `throw new Error(Unrecognised token: ${token})`.

Reply:
What will be the value of `stack[1]` in this test case?

Reply:
Another `(`.

Reply:
I'm more and more convinced that there is a better way to solve this though given that it doesn't seem super intuitive.

Reply:
I see, so `stack[1]` could be a nested function, but if not then it's a syntactic error... That's fair enough! Although the tokenizer would perhaps have already have correctly tokenized any nested functions? In which case, it may enough to just check for existence of "orphan" parenthseses.

Reply:
Exactly! However I'm not sure  what you mean by orphan, as (as you can see in the test case above) all brackets are matched, just not correctly.

Reply:
I meant "without a parent" which would be `max` in this case

So in this block, we have already parsed a `,`, which means we are parsing function args, and we want to confirm that any nested functions are correctly provided. The only valid value in `stack[1]` would be a function name. Other values that may possibly have passed through the tokenizer validation and exist in `stack[1]`, but are incorrect in this specific context, are:

* `(`

Any others? If not, perhaps you should just check `if (stack[0] !== '(' || stack[1] === '(')`

Reply:
> we have already parsed a `,`, which means we are parsing function args

Just thought of this, but I think this is the only place that this is checked for, so that has to be a valid use case (I'll add a test for that as well, e.g. `1 + (, 1)` would pass that test but fail the original)
> Any others?
As above, the stack would be `[ '(', '+' ]`, and in fact any other valid token could be there.

> any nested functions are correctly provided

I'm not sure what you mean by this? 

e.g. If the input is `validFunc( max( 1 , 2 ) , 3 )`:
At the first comma, stack is: `[ '(', 'max', '(', 'validFunc' ]`
As you can see, the nesting does not affect the parsing of this comma as all of the wrapping function things are further up the stack

At the second comma, stack is: `[ '(', 'validFunc' ]`
So there should be no issues there.

Reply:
> As above, the stack would be [ '(', '+' ], and in fact any other valid token could be there


That answers my question actually, so we can leave this as is! If you want to make this clearer you can extract the `FUNCTION_NAMES.includes(stack[1])` call to a local `isFunctionToken` method, to empasise that we check whether the token is a function name syntactically, not whether it's a function we support (which as you commented, would have  been caught at an earlier stage)

Reply:
Good idea, I think it's also worth adding a comment and some extra tests as suggested in this thread